src/V3Randomize.cpp

```diff
@@ -1722,55 +1827,89 @@ class RandomizeVisitor final : public VNVisitor {
         VL_DO_DANGLING(pushDeletep(nodep), nodep);
     }
     void visit(AstNodeFTaskRef* nodep) override {
-        if (nodep->name() == "rand_mode") {
-            AstMethodCall* const methodCallp = VN_CAST(nodep, MethodCall);
-            AstNodeExpr* const fromp = methodCallp ? methodCallp->fromp() : nullptr;
-            const RandModeTarget randModeTarget = RandModeTarget::get(fromp, m_modp);
-            UASSERT_OBJ(randModeTarget.classp, nodep,
-                        "Should have checked in RandomizeMarkVisitor");
-            AstVar* const randModeVarp = getRandModeVar(randModeTarget.classp);
-            AstNodeExpr* lhsp = nullptr;
-            if (randModeTarget.classp == m_modp) {
+        // Common logic for rand_mode and constraint_mode
+        const auto makeModeAssignLhs
```

Comment:
Enclosing function is large, prefer make this into separate member function. Similar below.